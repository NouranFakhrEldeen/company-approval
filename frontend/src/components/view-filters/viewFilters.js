import React from 'react';
import { formatDate } from '../../helpers';

const viewFilters = ({
  searchInputValue,
  iconSearch,
  clearTextSearch,
  // dateFilteration,
  toDate,
  fromDate,
  inDeviationTime,
}) => {


  const FilterationDiv = (value, id) => {
    return <div className={'search-label-filteration'}>
      <div
        className='d-inline-block light bg-light-dark border-radius-12 margin-5 vertical-align-middle'>
        <span className='padding-5 padding-bottom-0 d-inline-block vertical-align-top'>
          <span className={iconSearch} />
        </span>
        <span
          className='padding-3-5 padding-top-3 d-inline-block vertical-align-top'>
          {value}
        </span>
        <span
          className={`padding-3-5 padding-top-3 d-inline-block
               vertical-align-top padding-right-10 border-left-1 border-white border-solid
               cursor-pointer-hover filter-icon-hover `}
          onClick={() => { clearTextSearch(value, id); }}>&#10005;</span>
      </div>


    </div>;
  };
  return (

    <div className='row margin-bottom-10'>
      {searchInputValue && FilterationDiv(searchInputValue, 'searchTerm')}

      {fromDate && FilterationDiv(formatDate('dd-mm-yyyy', fromDate), 'from')}

      {toDate && FilterationDiv(formatDate('dd-mm-yyyy', toDate), 'to') }
      {inDeviationTime && FilterationDiv(formatDate('dd-mm-yyyy', inDeviationTime), 'inDeviationTime') }

    </div>
  );
};

export const ViewFilters = viewFilters;
