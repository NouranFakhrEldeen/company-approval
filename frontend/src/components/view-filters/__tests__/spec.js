
import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import { ViewFilters } from '../viewFilters';
describe('filter deviations form component', () => {
  it('Should render (smoke test)', () => {

    const { asFragment } = render(
      <Router>
        <ViewFilters
          filters={{}}
          t={key=>key}
          handleRemoveFilter={() => { }}
        />
      </Router>,
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});