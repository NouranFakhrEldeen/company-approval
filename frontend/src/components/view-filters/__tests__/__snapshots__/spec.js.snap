// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter deviations form component Should render (smoke test) 1`] = `
<DocumentFragment>
  <div
    class="row margin-bottom-10"
  />
</DocumentFragment>
`;
